upath/core.py

```diff
@@ -204,10 +235,10 @@ def parent(self):
     def stat(self):
         return self._accessor.stat(self)
 
-    def samefile(self, other_path):
+    def samefile(self, other_path) -> bool:
         raise NotImplementedError
 
-    def iterdir(self):
+    def iterdir(self: PT) -> Generator[PT, None, None]:
```

Comment:
Do you see an advantage in annotating this as a generator?

```suggestion
    def iterdir(self: PT) -> Iterator[PT]:
```

Reply:
mypy complains if this is annotated as Iterator:

```
upath/core.py:286: error: Return type "Iterator[UPath]" of "glob" incompatible with return type "Generator[UPath, None, None]" in supertype "Path"
upath/core.py:293: error: Return type "Iterator[UPath]" of "rglob" incompatible with return type "Generator[UPath, None, None]" in supertype "Path"
```